rest/database_error.go

```diff
@@ -0,0 +1,46 @@
+//  Copyright 2025-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package rest
+
+// DatabaseError denotes an error that occurred during database startup
+type DatabaseError struct {
+	ErrMsg string            `json:"error_message"`
+	Code   DatabaseErrorType `json:"error_code"`
+}
+
+type DatabaseErrorType uint8
+
+const (
+	DatabaseBucketConnectionError DatabaseErrorType = iota
```

Comment:
The `iota` pattern isn't great for things that have external consumers, since it's really easy to add or remove an entry in the middle of the list and change the subsequent error codes.

Think we'll just have to declare all of these codes explicitly to be safe.

Reply:
Just pushed up commit to have global map for error coder to message, I feel this fixes the issue of adding/removing error codes keeping that unique 1-1 mapping? 

Or do you think actually having to define error code and message constants separate would be best?